src/test/java/org/ays/auth/payload/RoleCreatePayload.java

```diff
@@ -21,4 +25,50 @@ public static RoleCreatePayload generate() {
         return roleCreatePayload;
     }
 
+    public static RoleCreatePayload generateRoleWithPermissions(Object permissions) {
+        RoleCreatePayload roleCreatePayload = new RoleCreatePayload();
+        roleCreatePayload.setName(AysRandomUtil.generateFirstName() + " Rol " + AysRandomUtil.generateAlphaSuffix());
+
+        List<String> permissionIds;
+
+        if (permissions instanceof Permission) {
+            permissionIds = Arrays.asList(PermissionDataSource.findPermissionIdByPermissionName(((Permission) permissions)));
+        } else if (permissions instanceof List) {
+            permissionIds = ((List<Permission>) permissions).stream()
+                    .map(PermissionDataSource::findPermissionIdByPermissionName)
+                    .collect(Collectors.toList());
+        } else {
+            throw new IllegalArgumentException("Permissions should be of type Permission or List<Permission>");
+        }
+
+        roleCreatePayload.setPermissionIds(permissionIds);
+
+        return roleCreatePayload;
+    }
+
+    public static RoleCreatePayload generateRoleWithCategoryPermissions(Object categories) {
+        RoleCreatePayload roleCreatePayload = new RoleCreatePayload();
+        roleCreatePayload.setName(AysRandomUtil.generateFirstName() + " Rol " + AysRandomUtil.generateAlphaSuffix());
+
+        List<String> permissionIds;
+
+        if (categories instanceof PermissionCategory) {
+            permissionIds = Arrays.stream(((PermissionCategory) categories).getPermissions())
+                    .map(PermissionDataSource::findPermissionIdByPermissionName)
+                    .collect(Collectors.toList());
+        } else if (categories instanceof List) {
+            permissionIds = ((List<PermissionCategory>) categories).stream()
+                    .flatMap(category -> Arrays.stream(category.getPermissions()))
+                    .map(PermissionDataSource::findPermissionIdByPermissionName)
+                    .distinct()
+                    .collect(Collectors.toList());
+        } else {
+            throw new IllegalArgumentException("Categories should be of type PermissionCategory or List<PermissionCategory>");
+        }
+
+        roleCreatePayload.setPermissionIds(permissionIds);
+
+        return roleCreatePayload;
+    }
```

Comment:
Buraya database fonksiyonları ve çokça kod eklememiz çok doğru bir yaklaşım olmamış. 

Bunun yerine;
- Bu metotları bu sınıflarda kaldırıp, permission ID'leri ilgili testte bulup buraya göndermek daha doğru bir seçenek.
- Ek olarak burada Instance of kontrollerine doğrudan ihtiyacımız yok doğrudan Category'leri PermissionDataSource'a gönderebiliriz. Bu sayede if-else'lerden kurtulmuş oluruz ✅